### PR TITLE
Add docs for `helm_extension` mechanism

### PR DESCRIPTION
**Pull request by viniciusdc**
## What does this implement/fix?

Includes doc section for helm_extension mechanism that was added as part of https://github.com/nebari-dev/nebari/issues/865

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

## Any other comments?

**Maybe this is already in the docs, and I just didn't find it; if that's the case, feel free to close it.

### CODE REVIEW

File: docs/docs/explanations/advanced-custom-settings.md [hunk 1]
```diff
@@ -211,3 +210,55 @@ As the name suggests, since it's a virtual private network, you need to be insid
 One way to achieve this is by creating a Virtual Machine (VM) inside the virtual network.
 Select the virtual network and subnet name under the networking settings of your cloud provider while creating the VM
 and then follow the usual deployment instructions as you would deploy from your local machine.
+
+#### Helm Extensions
+
+Nebari provides a way for any user to expand the infrastructure availablke by default by using the `helm_extensions` attribute, which allows for the management and customization of Kubernetes applications through Helm charts. The helm_extensions is a configuration construct that specifies a list of Helm charts and their respective settings.
```
Comment (marcelovilla) [inline docs/docs/explanations/advanced-custom-settings.md:216]: There's a typo on available

**Review by marcelovilla — CHANGES_REQUESTED** [review]

**Review by marcelovilla — APPROVED** [review]
LGTM! Thanks @viniciusdc!

File: docs/docs/explanations/advanced-custom-settings.md [hunk 2]
```diff
@@ -276,6 +276,61 @@ One way to achieve this is by creating a Virtual Machine (VM) inside the virtual
 Select the virtual network and subnet name under the networking settings of your cloud provider while creating the VM
 and then follow the usual deployment instructions as you would deploy from your local machine.
 
+:::note Note
+Current `conda_store.worker_resources` defaults are set at the minimum recommended resources for conda-store-workers - (conda-store [docs](https://conda.store/conda-store/references/faq#what-are-the-resource-requirements-for-conda-store-server))
+:::
```
Comment (viniciusdc) [inline docs/docs/explanations/advanced-custom-settings.md:281]:
```suggestion
```

File: docs/docs/explanations/advanced-custom-settings.md [hunk 3]
```diff
@@ -288,7 +343,3 @@ conda_store:
       cpu: 1
       memory: 4Gi
 ```
```
Comment (viniciusdc) [inline docs/docs/explanations/advanced-custom-settings.md:345]:
```suggestion
```

:::note Note
Current `conda_store.worker_resources` defaults are set at the minimum recommended resources for conda-store-workers - (conda-store [docs](https://conda.store/conda-store/references/faq#what-are-the-resource-requirements-for-conda-store-server))
:::
```

**Pull request merged**